app/scripts/controllers/permissions/index.js

```diff
@@ -267,6 +267,21 @@ export class PermissionsController {
     this._removePendingApproval(id)
   }
 
+  /**
+   * Rejects all pending permissions requests
+   * (Used in the event of closing a locked request window)
+   */
+  async rejectAllPermissionsRequests () {
+    if (!this.pendingApprovals.size) {
+      log.debug(`No permissions request in queue`)
+      return
+    }
+
+    this.pendingApprovals.forEach(async (_approval, id) => {
+      await this.rejectPermissionsRequest(id)
+    })
```

Comment:
Assuming `rejectAllPermissionsRequests` is called with `await`, this should properly bubble a Promise up the call chain that can be awaited to ensure all permission requests are rejected

```suggestion

    for (let id = 0; i < this.pendingApprovals.length; id++) {
      await this.rejectPermissionsRequest(id);
    }
```